pkg/resources/usercluster/deployment.go

```diff
@@ -307,3 +313,46 @@ func getLabelsArgValue(cluster *kubermaticv1.Cluster) (string, error) {
 	}
 	return string(bytes), nil
 }
+
+func getCloudProviderName(cluster *kubermaticv1.Cluster) (string, error) {
+	if cluster.Spec.Cloud.VSphere != nil {
+		return provider.VSphereCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.AWS != nil {
+		return provider.AWSCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.Openstack != nil {
+		return provider.OpenstackCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.GCP != nil {
+		return provider.GCPCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.Alibaba != nil {
+		return provider.AlibabaCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.Anexia != nil {
+		return provider.AnexiaCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.Azure != nil {
+		return provider.AzureCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.Digitalocean != nil {
+		return provider.DigitaloceanCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.Hetzner != nil {
+		return provider.HetznerCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.Kubevirt != nil {
+		return provider.KubevirtCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.Packet != nil {
+		return provider.PacketCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.BringYourOwn != nil {
+		return provider.BringYourOwnCloudProvider, nil
+	}
+	if cluster.Spec.Cloud.Fake != nil {
+		return provider.FakeCloudProvider, nil
+	}
+	return "", fmt.Errorf("provider unknown")
+}
```

Comment:
Why are we now using this function instead of `data.GetKubernetesCloudProviderName()`? The original function has been handling some additional cases including CCM/CSI migration.

Reply:
Because the function you mentioned doesn't properly return the provider in which the cluster is running but instead returns "external" as cloud provider when the `externalCloudProvider` is used. This function, instead simply returns the name of the provider, and gives it to the `user-cluster-controller-manager` as a flag. Besides, That flag appeared to be not used by the `user-cluster-controller-manager` itself before I introduced these changes.

Reply:
Okay, this makes sense, but I'd propose that we add this function to the same place as the `data.GetKubernetesCloudProviderName()` function, so it can be easily reused. Also, make sure to update the interface here: https://github.com/kubermatic/kubermatic/blob/ecc36450f80ab81a6d6e658dd7cba94b116233db/pkg/resources/usercluster/deployment.go#L57-L72